src/vm/frames.cpp

```diff
@@ -1238,7 +1238,16 @@ void TransitionFrame::PromoteCallerStack(promote_func* fn, ScanContext* sc)
     //If not "vararg" calling convention, assume "default" calling convention
     if (!MetaSig::IsVarArg(pFunction->GetModule(), callSignature))
     {
-        MetaSig msig(pFunction);
+        SigTypeContext typeContext(pFunction);
+        PCCOR_SIGNATURE pSig;
+        DWORD cbSigSize;
+        pFunction->GetSig(&pSig, &cbSigSize);
+
+        MetaSig msig(pSig, cbSigSize, pFunction->GetModule(), &typeContext);
```

Comment:
Do we know a matching fix for `PromoteCallerStackUsingGCRefMap` - adjust the GCRef map that gets saved into R2R image at crossgen time?

You should be able to verify it by compiling the affected tests using crossgen before running them.

Reply:
Yeah, it didn't work. Thanks for spotting! I pushed out a fix.